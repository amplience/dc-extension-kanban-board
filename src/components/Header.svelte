<script>
  import Divider from './Divider.svelte';

  export let contentItemsCount: number = 0;
  export let contentItemsPath: string = '';
</script>

<style lang="scss">
  header {
    margin: 0 8px 0 8px;
    height: 60px;
    display: flex;
    flex-direction: column;
    justify-content: center;
  }

  .info {
    display: flex;
    height: 100%;
    align-items: center;
  }

  .label {
    font-size: 13px;
    color: #999;
  }

  .value {
    font-size: 16px;
    color: #333;
  }
</style>

<header>
  <div class="info">
    <div class="info__count">
      <span class="value">{contentItemsCount}</span>
      <span class="label">content items</span>
    </div>
    <Divider vertical="true" />
    <div>
      <span class="label">Repo / folder</span>
      <span class="value">{contentItemsPath || 'none'}</span>
    </div>
  </div>
  <Divider />
</header>
