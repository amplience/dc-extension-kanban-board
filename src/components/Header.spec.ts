import { render } from '@testing-library/svelte';
import Header from './Header.svelte';

describe('header component', () => {
  it('should render a header component with values', () => {
      const { container } = render(Header, {
        contentItemsCount: 100,
        contentItemsPath: 'CONTENT_ITEMS_PATH'
      });

      expect(container.firstChild).toMatchSnapshot();
  });


  it('should render a header component with default values', () => {
    const { container } = render(Header, {
      // no-op.
    });

    expect(container.firstChild).toMatchSnapshot();
  });
});