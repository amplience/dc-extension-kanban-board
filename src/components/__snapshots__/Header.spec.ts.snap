// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`header component should render a header component with default values 1`] = `
<div>
  <header
    class="svelte-1pkjfof"
  >
    <div
      class="info svelte-1pkjfof"
    >
      <div
        class="info__count"
      >
        <span
          class="value svelte-1pkjfof"
        >
          0
        </span>
         
        <span
          class="label svelte-1pkjfof"
        >
          content items
        </span>
      </div>
       
      <div
        class="divider vertical svelte-mi6et7"
      />
       
      <div
        class="repo-folder-wrap svelte-1pkjfof"
      >
        <span
          class="label repo-folder svelte-1pkjfof"
        >
          Repo / folder
        </span>
         
        <span
          class="value path svelte-1pkjfof"
          title="all"
        >
          all
        </span>
      </div>
    </div>
     
    <div
      class="divider horizontal svelte-mi6et7"
    />
  </header>
</div>
`;

exports[`header component should render a header component with values 1`] = `
<div>
  <header
    class="svelte-1pkjfof"
  >
    <div
      class="info svelte-1pkjfof"
    >
      <div
        class="info__count"
      >
        <span
          class="value svelte-1pkjfof"
        >
          0
        </span>
         
        <span
          class="label svelte-1pkjfof"
        >
          content items
        </span>
      </div>
       
      <div
        class="divider vertical svelte-mi6et7"
      />
       
      <div
        class="repo-folder-wrap svelte-1pkjfof"
      >
        <span
          class="label repo-folder svelte-1pkjfof"
        >
          Repo / folder
        </span>
         
        <span
          class="value path svelte-1pkjfof"
          title="CONTENT_ITEMS_PATH"
        >
          CONTENT_ITEMS_PATH
        </span>
      </div>
    </div>
     
    <div
      class="divider horizontal svelte-mi6et7"
    />
  </header>
</div>
`;
